src/mono/browser/runtime/profiler.ts

```diff
@@ -82,26 +88,20 @@ export function endMeasure (start: TimeStamp, block: string, id?: string) {
     }
 }
 
-const stackFrames: number[] = [];
-export function mono_wasm_profiler_enter (): void {
-    if (runtimeHelpers.enablePerfMeasure) {
-        stackFrames.push(globalThis.performance.now());
-    }
+export function mono_wasm_profiler_now (): number {
+    return globalThis.performance.now();
 }
 
 const methodNames: Map<number, string> = new Map();
-export function mono_wasm_profiler_leave (method: MonoMethod): void {
-    if (runtimeHelpers.enablePerfMeasure) {
-        const start = stackFrames.pop();
-        const options = ENVIRONMENT_IS_WEB
-            ? { start: start }
-            : { startTime: start };
-        let methodName = methodNames.get(method as any);
-        if (!methodName) {
-            const chars = cwraps.mono_wasm_method_get_name(method);
-            methodName = utf8ToString(chars);
-            methodNames.set(method as any, methodName);
-        }
-        globalThis.performance.measure(methodName, options);
+export function mono_wasm_profiler_record (method: MonoMethod, start: number): void {
+    const options = ENVIRONMENT_IS_WEB
```

Comment:
Comment explaining this would be nice but isn't mandatory

Reply:
NodeJS has different API